pkg/reconciler/mutatingwebhook/reconcile.go

```
@@ -0,0 +1,82 @@
+}
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations,verbs=get;list;watch;update;
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations/status,verbs=get;update;patch
+
+// Reconcile is the reconciliation method for OSM MutatingWebhookConfiguration.
+func (r *MutatingWebhookConfigrationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	instance := &v1beta1.MutatingWebhookConfiguration{}
+
+	if err := r.Get(ctx, req.NamespacedName, instance); err != nil {
+		log.Error().Err(err).Msgf("failure reading object %s ", req.NamespacedName.String())
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// reconcile only for OSM mutatingWebhookConfiguration
+	var shouldUpdate bool
+	if r.OsmWebhook == instance.Name {
+		//check if CA bundle exists on webhook
+		for idx, webhook := range instance.Webhooks {
+			// CA bundle missing for webhook, update webhook to include the latest CA bundle
+			if webhook.Name == injector.MutatingWebhookName && webhook.ClientConfig.CABundle == nil {
+				log.Trace().Msgf("CA bundle missing for webhook : %s ", req.Name)
+				shouldUpdate = true
+				cn := certificate.CommonName(fmt.Sprintf("%s.%s.svc", constants.OSMControllerName, r.OsmNamespace))
+				cert, err := r.CertManager.GetCertificate(cn)
+				if err != nil {
+					return ctrl.Result{}, errors.Errorf("Error updating mutating webhook, unable to get certificate for the mutating webhook %s: %+s", req.Name, err)
+				}
+				instance.Webhooks[idx].ClientConfig.CABundle = cert.GetCertificateChain()
+			}
+		}
+	}
+
+	if !shouldUpdate {
+		log.Trace().Msgf("Mutatingwebhookconfiguration %s already compliant", req.Name)
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.Update(ctx, instance); err != nil {
+		log.Error().Err(err).Msgf("Error updating mutatingwebhookconfiguration %s: %s", req.Name, err)
+		return ctrl.Result{Requeue: false}, nil
```

If the update fails, do we not want to retry?

---

We could update to retry, and that would help for transient issues but it there's an actual issue then we would be stuck in a reconcile loop. WDYT ?

---

If there is an error, then we should return the error so that it can be retried. 
```go
return ctrl.Result{}, client.IgnoreNotFound(err)
```

---

Do we also need to check if the secret is deleted?

---

I've updated the retry logic when the update fails. 

As of the secret being deleted, the GetCertificate handles that already so I think we should be okay